core/DataTable/Filter/AddColumnsProcessedMetricsGoal.php

```diff
@@ -58,6 +65,16 @@
  */
 class AddColumnsProcessedMetricsGoal extends AddColumnsProcessedMetrics
 {
+    /**
+     * Process metrics for entry page views
+     */
+    const GOALS_ENTRY_PAGES = -4;
+
+    /**
+     * Process for page views
+     */
+    const GOALS_PAGES = -3;
+
```

Comment:
I'm not yet sure if introducing those new constants might have any implications. If I understood the implementation correct, the filter might get either passed one of those constant values or the id of a specific goal. 
So when viewing the goal reports for a specific goal, its id will be passed so only the metrics of that goal are included. 
Introducing those new constants might kind of change that behavior for the goals by pages reports only, as they might always include all goals (even though they might not be displayed afterwards). Or did I get something wrong here?

Reply:
The standard page and entry page reports do not currently have a way to select a specific goal, so instead columns get added for all goals. To support this the page / entry page constants always include all goals. Maybe in a future change we could add support to choose a specific goal on the page reports and revisit this?